### DISCUSSION TITLE
온도계와 속도계 데이터 수집 방법

### DISCUSSION THREAD

**Discussion by sapsareee** [luftaquila/monolith#19] [Q&A] [answered]
안녕하세요! 모노리스를 사용하려고 알아보고 있다가 질문이 생겨서 문의사항을 남깁니다!

1. 부품의 온도를 측정하고 싶습니다. 적외선 온도 센서에서 온 정보를 TMA-1에 바로 전달 해야하는지 아니면 아두이노 같은 다른 보드를 거쳐서 값을 받아와야 하는지 알고싶습니다. 속도계도 이런 방식인지 궁금합니다.

2. 이 모노리스 프로젝트를 제작할 당시에 당시 팀에서는 속도와 온도는 어떻게 측정했는지 여쭤보고싶습니다. 예를 들어서 어떤 장치를 사용했고 데이터를 어떻게 PCB 기판으로 옮겨오셨는지 궁금합니다.

**Comment by luftaquila** [comment, marked as answer]
안녕하세요. 관심 가져 주셔서 감사합니다.

저희 팀은 차량 속도는 모터 컨트롤러가 CAN으로 내보내는 모터 RPM 데이터로 계산했습니다. 모터와 모터 컨트롤러 온도도 마찬가지로 컨트롤러 CAN을 통해서 받았고, 배터리와 BMS 온도는 BMS CAN으로 받았습니다.

모노리스에는 CAN 데이터 해석 기능이 구현되어 있어서, CAN으로 전송되는 데이터는 모두 기록할 수 있습니다.

사용하시려는 적외선 온도 센서가 아날로그 전압을 출력하는 방식이라면, 모노리스의 아날로그 입력 채널을 바로 사용하실 수 있습니다. 다만 조금 찾아보니 UART나 SPI로 디지털 출력을 내는 센서들이 많은 듯 한데요, 이런 방식의 센서는 온도 데이터를 CAN이나 아날로그 전압으로 변환해서 모노리스에 입력할 아두이노 등이 하나 더 필요합니다.

아니면 프로젝트를 포크하셔서 모노리스에 직접 온도 센서를 추가하셔도 됩니다. 모노리스가 사용하는 GPS 센서가 UART로 구현되어 있기 때문에, 참고하시면 어렵지 않게 구현하실 수 있을 것으로 생각됩니다.
